magma/lib/magma/query/predicate/model.rb

```diff
@@ -31,11 +31,19 @@ def initialize(question, model, *query_args)
       @model = model
       @filters = []
 
-      if question.restrict? && @model.has_attribute?(:restricted)
-        # the model can be restricted, and we should withhold restricted data
-        query_args.unshift(
-          [ 'restricted', '::false' ]
-        )
+      if question.restrict?
+      # the model can be restricted, and we should withhold restricted data
+        restriction_model = model
+        ancestral_path = []
+        while restriction_model do
+          if restriction_model.has_attribute?(:restricted)
+            query_args.unshift(
```

Comment:
Should there be a break here?  We discussed that if you have multiple restricted in your heritage, that the nearest one takes presedence.